src/Controls/tests/DeviceTests/Stubs/WindowHandlerStub.iOS.cs

```diff
@@ -12,55 +12,84 @@ public class WindowHandlerStub : ElementHandler<IWindow, UIWindow>, IWindowHandl
 	{
 		TaskCompletionSource<bool> _finishedDisconnecting = new TaskCompletionSource<bool>();
 		public Task FinishedDisconnecting => _finishedDisconnecting.Task;
+		IView _currentView;
 
 		public static IPropertyMapper<IWindow, WindowHandlerStub> WindowMapper = new PropertyMapper<IWindow, WindowHandlerStub>(WindowHandler.Mapper)
 		{
 			[nameof(IWindow.Content)] = MapContent
 		};
 
-		private static void MapContent(WindowHandlerStub handler, IWindow window)
+		void UpdateContent(UIWindow platformView)
 		{
-			var view = window.Content.ToPlatform(handler.MauiContext);
-
-			if (window.Content is Shell)
+			CloseView(_currentView, platformView , () =>
 			{
-				var vc =
-					(window.Content.Handler as IPlatformViewHandler)
-						.ViewController;
+				var view = VirtualView.Content.ToPlatform(MauiContext);
+				_currentView = VirtualView.Content;
 
-				handler.PlatformView.RootViewController.PresentViewController(vc, false, null);
-			}
-			else
-			{
-				handler.PlatformView.RootViewController.View.AddSubview(view);
-			}
+				if (VirtualView.Content is Shell)
+				{
+					var vc =
+						(_currentView.Handler as IPlatformViewHandler)
+							.ViewController;
+
+					PlatformView.RootViewController.PresentViewController(vc, false, null);
+				}
+				else
+				{
+					PlatformView.RootViewController.View.AddSubview(view);
+				}
+			});
 		}
 
-		protected override void DisconnectHandler(UIWindow platformView)
+		void CloseView(IView view, UIWindow platformView, Action finishedClosing)
 		{
-			var vc = (VirtualView.Content.Handler as IPlatformViewHandler)
-							.ViewController;
+			if (view == null)
+			{
+				finishedClosing?.Invoke();
+				return;
+			}
+
+			var vc = (view.Handler as IPlatformViewHandler).ViewController;
 
-			if (VirtualView.Content is Shell)
+			if (view is Shell)
 			{
-				platformView.RootViewController
-					.PresentedViewController.
-					DismissViewController(false,
+				// This means shell never got to the point of being preesented
+				if (platformView
```

Comment:
Can we create a local here for PresentedViewController, just for clarity?
`var pvc = platformView?.RootViewController?.PresentedViewController;`?